modules/javafx.graphics/src/main/java/javafx/css/CssParser.java

```diff
@@ -4282,14 +4322,94 @@ private void parse(Stylesheet stylesheet, CssLexer lexer) {
                 return;
             }
 
-            stylesheet.getRules().add(new Rule(selectors, declarations));
+            stylesheet.getRules().add(new Rule(mediaRule, selectors, declarations));
 
             currentToken = nextToken(lexer);
 
+            while (expectedRBraces > 0) {
+                if (!consumeRBrace(lexer)) {
+                    return;
+                }
+
+                if (mediaRule != null) {
+                    mediaRule = mediaRule.getParent();
+                }
+
+                currentToken = nextToken(lexer);
+                expectedRBraces--;
```

Comment:
what happens in case of unbalanced braces and `expectedRBraces` < 0 ?

Reply:
The behavior is the same as with a normal unbalanced curly brace. I've added a test for it.